### PR TITLE
Add simple sync wrapper client

### PR DESCRIPTION
**Pull request by bratseth**
@vekterli please review

This adds one new class - SyncFeedClient - which provides a simple API for those who just have a List of document operations they want to feed reasonably fast without dealing with any complexity.

The other edits are nonfunctional cleanup only.

### CODE REVIEW

**Review by vekterli — APPROVED** [review]
:+1: a couple of minor comments

File: vespa-http-client/src/main/java/com/yahoo/vespa/http/client/SyncFeedClient.java [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+        // The rest of the state of this is reset each time we call expectResultsOf
+
+        private int resultsReceived;
+        private Exception exception = null;
+
+        /**
+         * A map from document ids to their results. This is initially populated with null values to keep track of
+         * which responses we are waiting for.
+         */
+        private Map<String, Result> results = null;
```
[the remark this answers is not in the document]
Comment (bratseth) [reply vespa-http-client/src/main/java/com/yahoo/vespa/http/client/SyncFeedClient.java]: Good point, done

File: vespa-http-client/src/test/java/com/yahoo/vespa/http/client/SyncFeedClientTest.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+        assertTrue(result.isSuccess());
+        assertEquals(3, result.results().size());
+        assertNull(result.exception());
+        assertEquals("id::test::1", result.results().get(0).getDocumentId());
+        assertEquals("id::test::2", result.results().get(1).getDocumentId());
+        assertEquals("id::test::3", result.results().get(2).getDocumentId());
+    }
+
+}
```
[the remark this answers is not in the document]
Comment (bratseth) [reply vespa-http-client/src/test/java/com/yahoo/vespa/http/client/SyncFeedClientTest.java]: Yes, I actually do that - this method is called multiple times in the test.

File: vespa-http-client/src/test/java/com/yahoo/vespa/http/client/SyncFeedClientTest.java [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    private void assertFeedSuccessful(SyncFeedClient feedClient) {
+        List<SyncOperation> operations = new ArrayList<>();
+
+        operations.add(new SyncOperation("id::test::1",
+                                         "{" +
+                                                       "    \"put\": \"id::test::1\"," +
+                                                       "    \"fields\": {" +
+                                                       "        \"title\": \"Title 1\"" +
+                                                       "    }" +
+                                                       "}"));
```
[the remark this answers is not in the document]
Comment (bratseth) [reply vespa-http-client/src/test/java/com/yahoo/vespa/http/client/SyncFeedClientTest.java]: Yes, it's a tradeoff with keeping it really simple though.

**Pull request merged**